docs/source/guide/build_pipelines/pipeline_parametrization.rst

```diff
@@ -0,0 +1,135 @@
+        artifacts["iris_model"] = copy.deepcopy(mod)
+        return artifacts
+
+    [...]
+
+    if __name__ == "__main__":
+        # Edit this section to customize the behavior of artifacts
+        artifacts = run_all_sessions()
+        print(artifacts)
+
+Instead, we could have run
+
+.. code-block:: python
+   :emphasize-lines: 10
+
+    # Load artifacts to use in pipeline building
+    preprocessing_art = lineapy.get("iris_preprocessed")
+    modeling_art = lineapy.get("iris_model")
+
+    # Build an Airflow pipeline using artifacts
+    lineapy.to_pipeline(
+        pipeline_name="iris_pipeline_parametrized",
+        artifacts=[preprocessing_art.name, modeling_art.name],
+        dependencies={modeling_art.name: {preprocessing_art.name}},
+        input_parameters=["url"],  # Specify variable(s) to parametrize
+        output_dir="./output/pipeline_parametrization/",
+        framework="AIRFLOW",
+    )
+
+to get a parametrized pipline, like so:
+
+.. code-block:: python
+   :emphasize-lines: 9, 20, 42
+
+    # ./output/pipeline_parametrization/iris_pipeline_parametrized_module.py
+
+    import argparse
+
+    import pandas as pd
+    from sklearn.linear_model import LinearRegression
+
+
+    def get_iris_preprocessed(url):
+        df = pd.read_csv(url)
+        color_map = {"Setosa": "green", "Versicolor": "blue", "Virginica": "red"}
+        df["variety_color"] = df["variety"].map(color_map)
+        df["d_versicolor"] = df["variety"].apply(lambda x: 1 if x == "Versicolor" else 0)
+        df["d_virginica"] = df["variety"].apply(lambda x: 1 if x == "Virginica" else 0)
+        return df
+
+    [...]
+
+    def run_session_including_iris_preprocessed(
+        url="https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv",
+    ):
+        # Given multiple artifacts, we need to save each right after
+        # its calculation to protect from any irrelevant downstream
+        # mutations (e.g., inside other artifact calculations)
+        import copy
+
+        artifacts = dict()
+        df = get_iris_preprocessed(url)
+        artifacts["iris_preprocessed"] = copy.deepcopy(df)
+        mod = get_iris_model(df)
+        artifacts["iris_model"] = copy.deepcopy(mod)
+        return artifacts
+
+    [...]
+
+    if __name__ == "__main__":
+        # Edit this section to customize the behavior of artifacts
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--url",
+            type=str,
+            default="https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv",
+        )
+        args = parser.parse_args()
+        artifacts = run_all_sessions(
+            url=args.url,
+        )
+        print(artifacts)
+
+As shown, we now have ``url`` factored out as an easily tunable parameter for the pipeline,
+which allows us to run it with various data sources beyond those we started with (hence increasing the
+pipeline's utility).
```

Comment:
maybe also include a sample of how airflow DAG is also modified. that way the readers will know how an airflow pipeline might become reusable with this change.

Reply:
Great idea, working on it